lighthouse-core/audits/metrics/first-contentful-paint.js

```diff
@@ -54,12 +64,13 @@ class FirstContentfulPaint extends Audit {
     const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
     const metricComputationData = {trace, devtoolsLog, settings: context.settings};
     const metricResult = await ComputedFcp.request(metricComputationData, context);
+    const scoreOptions = context.options || this.getDefaultScoreOptions(artifacts, context);
```

Comment:
can we have this be performed by the runner? i.e. check if default options is a function that accepts arguments and invoke it for the default options and/or permanently convert it to a function instead of getter property?

I like this idea of passing in context to determine default options overall though :)

Reply:
> permanently convert it to a function instead of getter property?

That was another approach I started with. @paulirish and I thought it best to avoid changing the interface for default options, but I didn't consider supporting both a function and the (current) getter property, which I like.

Reply:
> thought it best to avoid changing the interface for default options

If we had a bunch of users of our default options already I'd agree, but we're still in the early stages there so if we'd like to move IMO now would be the time rather than indefinitely support two modes :)

Reply:
> That was another approach I started with. @paulirish and I thought it best to avoid changing the interface for default options, but I didn't consider supporting both a function and the (current) getter property, which I like.

> If we had a bunch of users of our default options already I'd agree, but we're still in the early stages there so if we'd like to move IMO now would be the time rather than indefinitely support two modes :)

having a
```js
static getDefaultOptions(context) {
  if (artifacts.TestedAsMobileDevice) {
    return {
      scorePODR: 2000,
      scoreMedian: 4000,
    }
  } else {
    return {
      scorePODR: 800,
      scoreMedian: 1600,
    }
  }
}
}

static async audit(artifacts, context) {
  const scoreOptions = context.options;
  // ...
}
```
seems ergonomically (and functionally) equivalent to something like
```js
static get defaultOptions() {
  return {
    mobile: {
      scorePODR: 2000,
      scoreMedian: 4000,
    },
    desktop: {
      scorePODR: 800,
      scoreMedian: 1600,
    },
  };
}

static async audit(artifacts, context) {
  const scoreOptions = context.options[TestedAsMobileDevice ? 'mobile' : 'desktop'];
  // ...
}
```
and avoids churn for anyone using audit options.

Reply:
but speaking of users of default options...is there any reason to keep score control points in the options? The main push for that was to be able to make a desktop config that defined its own score curves (#4873) to override the default ones, but if desktop scoring is being moved into the audits themselves, any reason to continue defining them circuitously like this instead of as regular properties?

Reply:
> any reason to continue defining them circuitously like this instead of as regular properties?

Keeping it in the options is easy and offers users a way to override our opinions on scoring. Seems valuable to me.

Reply:
> offers users a way to override our opinions on scoring

my understanding is that's increasingly less desirable from the official metric audits, and if users want to do that they should use the computed metrics in custom audits.

Reply:
tho I am assuming users actually use this

Reply:
> tho I am assuming _some_ user actually uses this

Looks like the only people using it are [folks that are reusing the desktop config](https://github.com/search?l=JavaScript&q=scorePODR+options+emulatedFormFactor&type=Code).



> but speaking of users of default options...is there any reason to keep score control points in the options? 

I was thinking the same thing.

I'm in favor of moving scoring out out of options. (Roughly reverting #4927)  If a user really wants different scoring, they can create a plugin that requires a core metric audit and then recomputes the score their way.

I'm happy to add friction there.  I want audit options to be used for all sorts of other things but core perf metric scoring doesn't seem like something that should be part of our extensibility story.

@patrickhulce how sad would you be?

Reply:
> Looks like the only people using it are folks that are reusing the desktop config.

A GitHub search would find any open source tools that use Lighthouse + these config options, but it doesn't tell us anything about how power users might be using configs in their infra to run Lighthouse.

Reply:
Setting it up with options allows folks to run with advanced throttling on different connection types and run with just a slightly modified config ala `lr-desktop-config`. To say that removing scoring from options adds friction feels like quite an understatement. If we removed, they would need to completely reimplement *every metric audit* just to have meaningful scores of any connection type outside our two blessed ones, which sounds like an effective way to completely kill testing of other connection types. 

We also originally structured it this way to offer metric values on different connection types. We had dreams once upon a time of offering multiple views within the same report and while no one has really talked about something like that recently, we get that for free when the inputs all come from options and it would be rather difficult to accomplish and/or require hard-coded copies without it. If #8374 had been able to land, I would have shipped my slow-3g performance plugin that does this :)

Nuking scoring in options kills both of these dreams, and I'm not personally convinced that a custom config isn't already enough friction to discourage it. AFAICT from my brief search there are exactly 0 usages of custom `scorePODR` in open source github where the values aren't just copied from our guidance or `lr-desktop-config`.

In summary, I would be very, very sad to see this die.

Reply:
okay. :)

we'll keep the scoring parameters in audit options.

Reply:
I don't think it's a huge burden to make users do `class Slow3gInteractive extends Interactive {static getScoreOptions {return myAlteredScoreOptions;}`. But we could also be rid of the (undocumented) passing of an audit's default options back into itself with a simple `const {options = this.defaultOptions} = context` in the audit and still have the override behavior without `runner` trickiness.

...but I can live with the status quo :)

But I would very much like to see no new `options` magic for something we all agree no one uses to do something complicated when we can do it in a simple way :)